Realm/ObjectServerTests/RLMObjectServerTests.m

```diff
@@ -318,6 +319,67 @@ - (void)testDeleteObjects {
     }
 }
 
+#pragma mark - Encryption
+
+/// If client B encrypts its synced Realm, client A should be able to access that Realm with a different encryption key.
+- (void)testEncryptedSyncedRealm {
+    NSURL *url = REALM_URL();
+    RLMSyncUser *user = [self logInUserForCredentials:[RLMObjectServerTests basicCredentialsWithName:ACCOUNT_NAME()
+                                                                                            register:self.isParent]
+                                               server:[RLMObjectServerTests authServerURL]];
+
+    NSData *key = RLMGenerateKey();
+    RLMRealm *realm = [self openRealmForURL:url user:user encryptionKey:key
+                                 stopPolicy:RLMSyncStopPolicyAfterChangesUploaded immediatelyBlock:nil];
+    if (self.isParent) {
+        CHECK_COUNT(0, SyncObject, realm);
+        RLMRunChildAndWait();
+        [self waitForDownloadsForUser:user realms:@[realm] realmURLs:@[url] expectedCounts:@[@3]];
+    } else {
+        // Add objects.
+        [self addSyncObjectsToRealm:realm descriptions:@[@"child-1", @"child-2", @"child-3"]];
+        [self waitForUploadsForUser:user url:url];
+        CHECK_COUNT(3, SyncObject, realm);
+    }
+}
+
+/// If an encrypted synced Realm is re-opened with the wrong key, throw an exception.
+- (void)testEncryptedSyncedRealmWrongKey {
+    NSURL *url = REALM_URL();
+    RLMSyncUser *user = [self logInUserForCredentials:[RLMObjectServerTests basicCredentialsWithName:ACCOUNT_NAME()
+                                                                                            register:self.isParent]
+                                               server:[RLMObjectServerTests authServerURL]];
+
+    if (self.isParent) {
+        NSString *path;
+        @autoreleasepool {
+            RLMRealm *realm = [self openRealmForURL:url user:user encryptionKey:RLMGenerateKey()
+                                         stopPolicy:RLMSyncStopPolicyImmediately immediatelyBlock:nil];
+            path = realm.configuration.pathOnDisk;
+            CHECK_COUNT(0, SyncObject, realm);
+            RLMRunChildAndWait();
+            [self waitForDownloadsForUser:user realms:@[realm] realmURLs:@[url] expectedCounts:@[@3]];
+        }
+        RLMRealmConfiguration *c = [RLMRealmConfiguration defaultConfiguration];
```

Comment:
This test should also test re-opening it as a synced Realm with the wrong encryption key.